pkg/kubelet/config/file_linux.go

```diff
@@ -151,3 +155,26 @@ func (s *sourceFile) consumeWatchEvent(e *watchEvent) error {
 	}
 	return nil
 }
+
+func modifyStaticPodForPinnedManagement(pod *v1.Pod) {
+	if !managed.IsPodManaged(pod) {
+		return
+	}
+	if pod.ObjectMeta.Annotations == nil {
+		pod.ObjectMeta.Annotations = make(map[string]string)
+	}
+	for _, container := range pod.Spec.Containers {
+		if container.Resources.Requests.Cpu().IsZero() {
+			continue
+		}
+		var resourceLimit string
+		if val, valid := container.Resources.Requests.Cpu().AsInt64(); valid {
+			resourceLimit = fmt.Sprintf("%v", val*1000)
+		} else {
+			resourceLimit = fmt.Sprintf("%v", container.Resources.Requests.Cpu().AsApproximateFloat64()*1000)
+		}
+		containerNameKey := fmt.Sprintf(managed.ContainerCpuAnnotationFormat, container.Name)
+		pod.ObjectMeta.Annotations[containerNameKey] = resourceLimit
```

Comment:
WDT about using `container.Resources.Requests.Cpu().String()`, and the CRI-O will use `resource.ParseQuantity()` to get the desired quantity. It also should reflect the value of annotation under the KEP - https://github.com/openshift/enhancements/pull/703/files#diff-bdf426b496fe2c7122db9eeaf303b30ff4561b4f7ac157c21996edff1e67e239R349